File: VB/Q257245/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace Q257245

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim list = New BindingList(Of Item)()
            For i As Integer = 0 To 10 - 1
                list.Add(New Item() With {.ID = i, .Name = "Name" & i})
            Next

            myGridControl1.DataSource = list
        End Sub
    End Class

    Public Class Item

        Public Property ID As Integer

        Public Property Name As String
    End Class
End Namespace

File: CS/Q257245/Form1.cs
using System;
using System.ComponentModel;
using System.Windows.Forms;

namespace Q257245
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            var list = new BindingList<Item>();
            for (int i = 0; i < 10; i++)
                list.Add(new Item() { ID = i, Name = "Name" + i });
            myGridControl1.DataSource = list;
        }
    }
    public class Item
    {
        public int ID { get; set; }
        public string Name { get; set; }
    }
}